### PR TITLE
Remove some nullable disable comments

### PR DESCRIPTION
**Pull request by stefannikolei**
#2231

Removed the nullable disable in all places where fixing it was an easy task

### CODE REVIEW

**Comment by stefannikolei** [comment]
After these only 10 files have the annotation `nullable disable`

File: src/ImageSharp/Advanced/AdvancedImageExtensions.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public static IImageEncoder DetectEncoder(this Image source, string filePath)
         Guard.NotNull(filePath, nameof(filePath));
 
         string ext = Path.GetExtension(filePath);
-        IImageFormat format = source.GetConfiguration().ImageFormatsManager.FindFormatByFileExtension(ext);
+        IImageFormat? format = source.GetConfiguration().ImageFormatsManager.FindFormatByFileExtension(ext);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Advanced/AdvancedImageExtensions.cs:30]: Ah... This is one of the areas I want to have a look at. Ideally this method should be rewritten to use the try-out pattern.

e.g 

``` c#
bool ImageFormatsManager.TryFindFormatByFileExtension(ext, [NotNullWhen(true)] out IImageFormat? format)
```

We'd return named tuples for async equivalents.

File: src/ImageSharp/Formats/Bmp/BmpImageFormatDetector.cs [hunk 2]
```diff
@@ -15,7 +14,7 @@ public sealed class BmpImageFormatDetector : IImageFormatDetector
     public int HeaderSize => 2;
 
     /// <inheritdoc/>
-    public IImageFormat DetectFormat(ReadOnlySpan<byte> header)
+    public IImageFormat? DetectFormat(ReadOnlySpan<byte> header)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Bmp/BmpImageFormatDetector.cs:17]: Same here. We'd convert to use `TryDetectFormat`

File: src/ImageSharp/Formats/ImageFormatManager.cs [hunk 3]
```diff
@@ -93,7 +92,7 @@ public void AddImageFormat(IImageFormat format)
     /// </summary>
     /// <param name="extension">The extension to discover</param>
     /// <returns>The <see cref="IImageFormat"/> if found otherwise null</returns>
-    public IImageFormat FindFormatByFileExtension(string extension)
+    public IImageFormat? FindFormatByFileExtension(string extension)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/ImageFormatManager.cs:95]: All these would use `TryXX`

File: src/ImageSharp/Advanced/AdvancedImageExtensions.cs (hunk 1 above)
Comment (stefannikolei) [reply src/ImageSharp/Advanced/AdvancedImageExtensions.cs:30]: Ok gonna rewrite that.

Comment (JimBobSquarePants) [reply src/ImageSharp/Advanced/AdvancedImageExtensions.cs:30]: Legend. Thanks! It's something I very much wish we'd done from the start.

File: src/ImageSharp/Formats/ImageFormatManager.cs [hunk 4]
```diff
@@ -91,8 +92,9 @@ public void AddImageFormat(IImageFormat format)
     /// For the specified file extensions type find the e <see cref="IImageFormat"/>.
     /// </summary>
     /// <param name="extension">The extension to discover</param>
-    /// <returns>The <see cref="IImageFormat"/> if found otherwise null</returns>
-    public IImageFormat? FindFormatByFileExtension(string extension)
+    /// <param name="format">The <see cref="IImageFormat"/> if found otherwise null</param>
+    /// <returns>False if no format was found</returns>
+    public bool TryFindFormatByFileExtension(string extension, [NotNullWhen(true)] out IImageFormat? format)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/ImageFormatManager.cs:97]: 👍

File: src/ImageSharp/Image.FromBytes.cs [hunk 5]
```diff
@@ -27,7 +26,7 @@ public static IImageFormat DetectFormat(ReadOnlySpan<byte> data)
     /// <param name="data">The byte span containing encoded image data to read the header from.</param>
     /// <exception cref="ArgumentNullException">The options are null.</exception>
     /// <returns>The mime type or null if none found.</returns>
-    public static IImageFormat DetectFormat(DecoderOptions options, ReadOnlySpan<byte> data)
+    public static IImageFormat? DetectFormat(DecoderOptions options, ReadOnlySpan<byte> data)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Image.FromBytes.cs:29]: We still need to convert this and above to use the Try...Out pattern

File: src/ImageSharp/Processing/Processors/Convolution/ConvolutionProcessorHelpers.cs [hunk 6]
```diff
@@ -99,7 +98,7 @@ internal static float[] CreateGaussianSharpenKernel(int size, float weight)
     /// <param name="row">The resulting 1D row vector, if possible.</param>
     /// <param name="column">The resulting 1D column vector, if possible.</param>
     /// <returns>Whether or not <paramref name="matrix"/> was linearly separable.</returns>
-    public static bool TryGetLinearlySeparableComponents(this DenseMatrix<float> matrix, out float[] row, out float[] column)
+    public static bool TryGetLinearlySeparableComponents(this DenseMatrix<float> matrix, out float[]? row, out float[]? column)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Processing/Processors/Convolution/ConvolutionProcessorHelpers.cs:101]: We can add the `NotNullWhen` attributes here

File: src/ImageSharp/Image.FromBytes.cs (hunk 5 above)
Comment (stefannikolei) [reply src/ImageSharp/Image.FromBytes.cs:29]: Do you want to convert InternalDetectFormat from Image.Decode.cs to the Try Pattern or just return IImageFormat?

**Comment by stefannikolei** [comment]
> After these only 10 files have the annotation `nullable disable`

Ok i have no clue how I searched. In total there a more nullable disable in the Solution.. Probably I searched while in a specific scope ... oO

File: src/ImageSharp/Image.FromBytes.cs (hunk 5 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/Image.FromBytes.cs]: Ideally it should use the Try pattern all the way down.

Comment (JimBobSquarePants) [reply src/ImageSharp/Image.FromBytes.cs]: Having a second look and we wouldn't be able to adopt the Try pattern there actually. The method is used within a lambda which doesn't allow ref or out parameters..

**Review by JimBobSquarePants — APPROVED** [review]
I think this is good to merge for now. Thanks!

I want to have a discussion about how to handle the various `Image.Load` APIs next.

**Pull request merged**